euphoria-core/src/main/java/cz/seznam/euphoria/core/client/io/Context.java

```diff
@@ -16,27 +16,8 @@
 package cz.seznam.euphoria.core.client.io;
 
 /**
- * A collector of elements. Used in functors.
- *
- * @param <T> the type of elements collected through this context
+ * Used in user defined functions to access environment methods.
  */
-public interface Context<T> {
-
-  /**
-   * Collects the given element to the output of this context.
-   *
-   * @param elem the element to collect
-   */
-  void collect(T elem);
-
-  /**
-   * Retrieves the window - if any - underlying the current
-   * execution of this context.
-   *
-   * @return {@code null} if this context is not executed within a
-   *          windowing strategy, otherwise the current window of
-   *          this context
-   */
-  Object getWindow();
+public interface Context extends Environment {
```

Comment:
This unfortunately implies that you will still have to specify the types of the lambda arguments, am I right? i.e.
```java
 FlatMap.of(...)
   .using((Integer in, Context ctx) -> {
   });
```
instead of just
```java
 FlatMap.of(...)
   .using((e, c) -> {
   });
```
Would it be possible to do something about that? I think that it would unfortunately mean that we would have to either add additional argument to the `UnaryFunctor`, or use other method name, e.g. `usingRich()`.
I'm aware that it would be more work. :-/
What do you think? And what do you think about naming the `Env` functions instead of `Env` suffix by `Rich` prefix?